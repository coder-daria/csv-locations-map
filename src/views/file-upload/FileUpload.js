import React, { memo } from 'react';
import { func } from 'prop-types';

import useFileUpload from './utils/use-file-upload';

import { Container, StyledFileUpload, ErrorMessage } from './FileUpload.styles';

function FileUpload({ uploadFile }) {
  const { onUpload, errorMessage } = useFileUpload({ uploadFile });

  return (
    <Container>
      <StyledFileUpload>
        <span>Upload file</span>
        <input accept=".csv" type="file" onChange={onUpload} />
      </StyledFileUpload>
      <ErrorMessage isVisible={!!errorMessage}>{errorMessage}</ErrorMessage>
    </Container>
  );
}

FileUpload.propTypes = {
  uploadFile: func.isRequired,
};

export default memo(FileUpload);
