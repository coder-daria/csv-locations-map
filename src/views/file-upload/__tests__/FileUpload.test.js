import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import FileUpload from '../FileUpload';

const setup = () => {
  const uploadFile = jest.fn();

  return shallow(<FileUpload uploadFile={uploadFile} />);
};

describe('<FileUpload />', () => {
  it('should match snapshot', () => {
    const wrapper = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
