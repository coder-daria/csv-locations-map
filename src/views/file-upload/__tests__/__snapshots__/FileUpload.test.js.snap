// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUpload /> should match the snapshot 1`] = `
<styled.div>
  <styled.div>
    <span>
      Upload file
    </span>
    <input
      accept=".csv"
      onChange={[Function]}
      type="file"
    />
  </styled.div>
  <div />
</styled.div>
`;
